src/plugins/data/public/query/filter_manager/filter_manager.ts

```diff
@@ -157,7 +154,16 @@ export class FilterManager {
     this.handleStateUpdate(newFilters);
   }
 
-  public setFilters(newFilters: esFilters.Filter[]) {
+  public setFilters(
+    newFilters: esFilters.Filter[],
+    pinFilterStatus: boolean = this.uiSettings.get('filters:pinnedByDefault')
+  ) {
+    const store = pinFilterStatus
+      ? esFilters.FilterStateStore.GLOBAL_STATE
+      : esFilters.FilterStateStore.APP_STATE;
+
+    FilterManager.setFiltersStore(newFilters, store);
```

Comment:
note: FilterManager.setFiltersStore() sets new `store` only if `filter` is missing it

Reply:
Maybe rename method?

Reply:
Agree that we should rename it. But I think we better do it in separate pr to keep this one scoped to filter_manager and just the bugfix which we backport to 7.6